gnd/src/main/java/com/google/android/gnd/repository/FeatureRepository.java

```diff
@@ -128,19 +128,32 @@ private Completable updateLocalFeature(RemoteDataEvent<Feature> event) {
   }
 
   private FeatureMutation fromFeature(Feature feature, Type type) {
-    Optional<Point> newLocation =
-        feature instanceof PointFeature
-            ? Optional.of(((PointFeature) feature).getPoint())
-            : Optional.empty();
-    return FeatureMutation.builder()
-        .setType(type)
-        .setProjectId(feature.getProject().getId())
-        .setFeatureId(feature.getId())
-        .setLayerId(feature.getLayer().getId())
-        .setNewLocation(newLocation)
-        .setUserId(authManager.getCurrentUser().getId())
-        .setClientTimestamp(new Date())
-        .build();
+    if (feature instanceof PointFeature) {
```

Comment:
[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself): Most of the logic in these two conditional branches is identical. Instead of repeated the logic twice, you can create the builder and set the comment fields outside to conditional, then set the remaining fields inside.

Reply:
set the other fields and then checking if its PointFeature then setPoint otherwise setPolygon